### PR TITLE
BREAKS: make MN HTTP compileOnly and it moves token classes to security

### PR DESCRIPTION
**Pull request by sdelamo**
This PR moves several classes from the `jwt` module to `security`. This change will allow Micronaut Security to support other tokens (for example, [Paseto tokens](https://github.com/micronaut-projects/micronaut-security/pull/813)) apart from JWT tokens.

It also makes Micronaut HTTP a compile only dependency. It changes several APIs to take a generic instead of an HTTP Request.

### CODE REVIEW

**Review by graemerocher — APPROVED** [review]

File: security/src/main/java/io/micronaut/security/endpoints/introspection/DefaultIntrospectionProcessor.java [hunk 1]
```diff
@@ -46,7 +45,7 @@
  * @since 2.1.0
  */
 @Singleton
-public class DefaultIntrospectionProcessor implements IntrospectionProcessor {
+public class DefaultIntrospectionProcessor<T> implements IntrospectionProcessor<T> {
```
Comment (timyates) [inline security/src/main/java/io/micronaut/security/endpoints/introspection/DefaultIntrospectionProcessor.java:48]: `<T>` needs javadoc

File: security-jwt/src/main/java/io/micronaut/security/token/jwt/validator/JwtValidator.java [hunk 2]
```diff
@@ -333,7 +333,7 @@ public static Builder builder() {
     /**
      * A builder for {@link JwtValidator}.
      */
-    public static final class Builder {
+    public static final class Builder<T> {
```
Comment (timyates) [inline security-jwt/src/main/java/io/micronaut/security/token/jwt/validator/JwtValidator.java:336]: `<T>` needs javadoc

File: security-ldap/src/main/java/io/micronaut/security/ldap/LdapAuthenticationProvider.java [hunk 3]
```diff
@@ -52,7 +51,7 @@
  * @author James Kleeh
  * @since 1.0
  */
-public class LdapAuthenticationProvider implements AuthenticationProvider, Closeable {
+public class LdapAuthenticationProvider<T> implements AuthenticationProvider<T>, Closeable {
```
Comment (timyates) [inline security-ldap/src/main/java/io/micronaut/security/ldap/LdapAuthenticationProvider.java:54]: `<T>` needs javadoc

**Review by timyates — APPROVED** [review]

File: security-ldap/src/main/java/io/micronaut/security/ldap/LdapAuthenticationProvider.java (hunk 3 above)
Comment (sdelamo) [reply security-ldap/src/main/java/io/micronaut/security/ldap/LdapAuthenticationProvider.java:55]: resolved via https://github.com/micronaut-projects/micronaut-security/pull/1320/commits/fdcc2b1f2519452727309838cec74511ccc1877c

File: security/src/main/java/io/micronaut/security/endpoints/introspection/DefaultIntrospectionProcessor.java (hunk 1 above)
Comment (sdelamo) [reply security/src/main/java/io/micronaut/security/endpoints/introspection/DefaultIntrospectionProcessor.java:49]: resolved via https://github.com/micronaut-projects/micronaut-security/pull/1320/commits/fdcc2b1f2519452727309838cec74511ccc1877c

**Pull request merged**